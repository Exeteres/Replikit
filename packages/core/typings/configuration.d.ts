import { RecursiveRequired, CoreConfiguration } from "@replikit/core/typings";

export interface Configuration {
    core?: CoreConfiguration;
}

export type ResolvedConfiguration = RecursiveRequired<Configuration>;
