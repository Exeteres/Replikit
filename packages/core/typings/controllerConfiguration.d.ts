export interface ControllerConfiguration {
    resolveAccountInfo?: boolean;
    resolveChannelInfo?: boolean;
}
