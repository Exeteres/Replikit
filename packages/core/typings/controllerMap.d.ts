// eslint-disable-next-line @typescript-eslint/no-empty-interface
export interface ControllerMap {}

export type ControllerName = keyof ControllerMap;
