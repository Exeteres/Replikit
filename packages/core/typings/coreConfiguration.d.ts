import { LogLevelName } from "@replikit/core/typings";

export interface CoreCacheConfiguration {
    expire?: number;
}

export interface CoreConfiguration {
    logLevel?: LogLevelName;
    disabledControllers?: string[];
    cache?: CoreCacheConfiguration;
}
