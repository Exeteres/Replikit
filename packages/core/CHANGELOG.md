# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.17.2](https://github.com/Exeteres/Replikit/compare/v0.17.1...v0.17.2) (2020-08-24)

**Note:** Version bump only for package @replikit/core





## [0.17.1](https://github.com/Exeteres/Replikit/compare/v0.17.0...v0.17.1) (2020-08-21)

**Note:** Version bump only for package @replikit/core





# [0.17.0](https://github.com/Exeteres/Replikit/compare/v0.16.0...v0.17.0) (2020-08-21)


### Features

* **core:** Add Identifier type ([e97ef90](https://github.com/Exeteres/Replikit/commit/e97ef90d24068fe4c16842b17bae0b8e6570ec47))





# [0.16.0](https://github.com/Exeteres/Replikit/compare/v0.15.1...v0.16.0) (2020-08-18)


### Bug Fixes

* **core:** Fix bug in text tokenization system ([43404bf](https://github.com/Exeteres/Replikit/commit/43404bffdb878e1f4abd18f69970cd6520251dd9))


### Features

* **core:** Add support for message headers ([5754c94](https://github.com/Exeteres/Replikit/commit/5754c94e0f89edb4ff3dde4fe410ade303bfb58f))





## [0.15.1](https://github.com/Exeteres/Replikit/compare/v0.15.0...v0.15.1) (2020-08-13)

**Note:** Version bump only for package @replikit/core





# [0.15.0](https://github.com/Exeteres/Replikit/compare/v0.14.0...v0.15.0) (2020-08-13)

**Note:** Version bump only for package @replikit/core





# [0.14.0](https://github.com/Exeteres/Replikit/compare/v0.13.0...v0.14.0) (2020-08-05)


### Bug Fixes

* **cli:** Fix typings ([9851fe5](https://github.com/Exeteres/Replikit/commit/9851fe5b87e07318da5c085f4b1bbddb23984136))





# [0.13.0](https://github.com/Exeteres/Replikit/compare/v0.12.1...v0.13.0) (2020-08-04)


### Bug Fixes

* **core:** Fix constructor type ([9687c48](https://github.com/Exeteres/Replikit/commit/9687c48773917919a2d8c38add27fba15cc11f00))


### Features

* **core:** Add ability to use pipe method in builders ([95c3cca](https://github.com/Exeteres/Replikit/commit/95c3cca0d1f4c5203fde3386c3775ecc77b92a1b))





# [0.12.0](https://github.com/Exeteres/Replikit/compare/v0.11.2...v0.12.0) (2020-07-30)


### Bug Fixes

* **core:** Rename query field to text ([4b0a930](https://github.com/Exeteres/Replikit/commit/4b0a930a4a840efcf56bdc6dfcf0b77a9430a30c))


### Features

* **core:** Refactor types and add support for inline queries ([2e52258](https://github.com/Exeteres/Replikit/commit/2e52258587cf0f2d31061c991eb2489ec7acda9e))





## [0.11.2](https://github.com/Exeteres/Replikit/compare/v0.11.1...v0.11.2) (2020-07-24)

**Note:** Version bump only for package @replikit/core





## [0.11.1](https://github.com/Exeteres/Replikit/compare/v0.11.0...v0.11.1) (2020-07-24)

**Note:** Version bump only for package @replikit/core






# [0.11.0](https://github.com/Exeteres/Replikit/compare/v0.10.1...v0.11.0) (2020-07-24)


### Bug Fixes

* **core:** Fix deepmerge ([132b184](https://github.com/Exeteres/Replikit/commit/132b18493e02603c195af70a2cfff0e9c31dc383))


### Features

* **sessions:** Add sessions package ([e7a0adf](https://github.com/Exeteres/Replikit/commit/e7a0adf3bbe35091bad6dcf28395e3c389a20a83))





## [0.10.1](https://github.com/Exeteres/Replikit/compare/v0.10.0...v0.10.1) (2020-06-30)

**Note:** Version bump only for package @replikit/core






# [0.10.0](https://github.com/Exeteres/Replikit/compare/v0.9.0...v0.10.0) (2020-06-29)

**Note:** Version bump only for package @replikit/core





# [0.9.0](https://github.com/Exeteres/Replikit/compare/v0.8.0...v0.9.0) (2020-06-28)


### Features

* Add advanced metadata processing ([54c7820](https://github.com/Exeteres/Replikit/commit/54c782021f5ebc16d784f210d273f8d7490d2af5))





# [0.8.0](https://github.com/Exeteres/Replikit/compare/v0.7.2...v0.8.0) (2020-06-27)


### Features

* **core:** Add core support for replies ([21f3318](https://github.com/Exeteres/Replikit/commit/21f3318f7bd031233856e7a1e1ddd29d73ef5f04))





## [0.7.2](https://github.com/Exeteres/Replikit/compare/v0.7.1...v0.7.2) (2020-06-23)

**Note:** Version bump only for package @replikit/core





# [0.7.0](https://github.com/Exeteres/Replikit/compare/v0.6.0...v0.7.0) (2020-06-23)

**Note:** Version bump only for package @replikit/core





# [0.6.0](https://github.com/Exeteres/Replikit/compare/v0.5.3...v0.6.0) (2020-06-21)

**Note:** Version bump only for package @replikit/core





## [0.5.3](https://github.com/Exeteres/Replikit/compare/v0.5.2...v0.5.3) (2020-06-20)

**Note:** Version bump only for package @replikit/core





## [0.5.2](https://github.com/Exeteres/Replikit/compare/v0.5.1...v0.5.2) (2020-06-20)

**Note:** Version bump only for package @replikit/core





# 0.5.0 (2020-06-20)


### Features

* **cli:** Add build command ([5c9dc57](https://github.com/Exeteres/Replikit/commit/5c9dc57b9ac2428f4970096dcfc6ae75b2e4dd27))






# [0.4.0](https://github.com/Exeteres/Replikit/compare/v0.3.0...v0.4.0) (2020-06-19)


### Features

* **cli:** Add build command ([0212d7f](https://github.com/Exeteres/Replikit/commit/0212d7f9c87d517b8cf220be9b32376551c3603b))





# [0.3.0](https://github.com/Exeteres/Replikit/compare/v0.2.0...v0.3.0) (2020-06-14)

**Note:** Version bump only for package @replikit/core





# [0.2.0](https://github.com/Exeteres/Replikit/compare/v0.1.0...v0.2.0) (2020-06-13)

**Note:** Version bump only for package @replikit/core
