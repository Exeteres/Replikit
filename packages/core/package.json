{
    "name": "@replikit/core",
    "version": "0.11.1",
    "license": "MIT",
    "main": "dist/index.js",
    "typings": "dist/index.d.ts",
    "files": [
        "dist",
        "typings"
    ],
    "scripts": {
        "build": "tsc -p tsconfig.build.json"
    }
}
