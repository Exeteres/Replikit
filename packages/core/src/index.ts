/// <reference path="../typings/index.d.ts" />

export * from "./models/channelType";
export * from "./models/attachmentType";
export * from "./models/textTokenKind";
export * from "./models/textTokenProp";
export * from "./models/logLevel";

export * from "./composition/compositionItem";
export * from "./composition/compositionFactory";
export * from "./composition/compositionContext";
export * from "./composition/composition";

export * from "./errors";
export * from "./utils";
export * from "./hookStorage";
export * from "./config";
export * from "./eventHandler";
export * from "./cacheManager";
export * from "./textTokenizer";
export * from "./textFormatter";
export * from "./controller";
export * from "./controllerStorage";

export * from "./logging/consoleLogProvider";
export * from "./logging/logManager";
export * from "./logging/logger";

import "./startup";
