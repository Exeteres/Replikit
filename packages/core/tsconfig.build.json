{
    "extends": "../../tsconfig.build.json",
    "compilerOptions": {
        "outDir": "dist",
        "paths": {
            "@replikit/core": ["core/src"]
        }
    },
    "include": ["src"]
}
