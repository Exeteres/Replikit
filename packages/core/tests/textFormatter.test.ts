import { TextFormatter, TextTokenKind, TextTokenProp } from "@replikit/core";
import { TextToken } from "@replikit/core/typings";

const formatter = new TextFormatter()
    .addVisitor(TextTokenKind.Mention, tokens => {
        return `[id${tokens.id}|${tokens.text ?? tokens.username}]`;
    })
    .addPropFormatter(TextTokenProp.Bold, "**")
    .addPropFormatter(TextTokenProp.Underline, "<u>", "</u>");

describe("TextFormatter", () => {
    it("should convert array of tokens to text", () => {
        const text: TextToken[] = [
            {
                kind: TextTokenKind.Text,
                text: "bold",
                props: [TextTokenProp.Bold]
            },
            { kind: TextTokenKind.Text, text: " ", props: [] },
            { kind: TextTokenKind.Mention, text: "text", id: 1, props: [] },
            {
                kind: TextTokenKind.Mention,
                username: "username",
                id: 2,
                props: [TextTokenProp.Underline]
            }
        ];

        const result = formatter.formatText(text);
        expect(result).toMatchSnapshot();
    });
});
