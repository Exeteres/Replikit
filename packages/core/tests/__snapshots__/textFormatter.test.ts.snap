// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFormatter should convert array of tokens to text 1`] = `"**bold** [id1|text]<u>[id2|username]</u>"`;
